### ISSUE TITLE
Allow to discuss and present ideas freely for core and/or plugin development

### ISSUE THREAD

**Issue by Xrayez** [godotengine/godot-proposals#47]
## The bar is too high just to spark ideas here

Continuing an off-topic discussion from https://github.com/godotengine/godot-proposals/issues/39#issuecomment-529095104.

*Note*: I'm personally against segregation but there seems to be no consensus with maintainers so I'm just seeking alternatives.

## Example repository
[godot-ideas](https://github.com/godot-extended-libraries/godot-ideas)

It mimics the process of `godot-proposals` without too much restrictions, and users are pointed to this repository if they feel like they could get approval from core developers, else it would be a nice place to discuss stuff without imposing too much rules.

## Problem
We've established previously that proposals could be classified as:

* Nice-to-have
* Need-to-have (aka must-have)

The "Nice-to-have" type of proposals, based on the official guide and feedback, are going to be closed eventually without them being considered for review, because allegedly they don't fulfill the actual needs and fix problems that users are dealing with on day-to-day basis.

While I completely understand the amount of "low-quality" proposals that are going to spawn if we waive the restriction of having to fill out every single question, the need to express ideas and be heard still exists, and we'd rather avoid dealing with frustrated users.

The listed community channels aren't as functional or reliable as they could be, see https://github.com/godotengine/godot-proposals/issues/47#issuecomment-529203868.

### Human factor 🙂

1. I guess people wouldn't like their proposals to be "closed" even if they can be reopened by maintainers if deemed satisfactory or needed, there's enough of examples how this makes people discouraged and rather frustrated.

2. The feeling of guilt that people feel starting random and general-purpose discussions by having to type "wall of text" godotengine/godot#19486. This can be avoided with dedicated place for that.

## Proposal
There should be an **_official_** repository for people to allow to discuss their ideas there. Maintainers are free to "ignore" these ideas so they don't really have to put much effort by having to read and understand them. Don't feel like users are **demanding** you to make it happen. Moreover, such repository would be a place for people to ask the questions like:

* Is it possible to make this as a plugin or should I go make a proposal to `godot-proposals`?
* I'm not sure whether this feature is needed at all, but perhaps it will in the future?

[Controversial topics](https://github.com/godotengine/godot/issues?q=is%3Aopen+is%3Aissue+label%3Adiscussion+label%3A%22feature+proposal%22+sort%3Acomments-desc) would be a perfect fit for this.

Such repository wouldn't necessary be targeted for Godot core, but a place for people to kind of request independent developers to write modules/plugins for them, but it needs to be centralized. An author is free to open and close an issue on his own.

Also see alternative approach/reasoning in https://github.com/godotengine/godot-proposals/issues/47#issuecomment-529388737.

<details>
  <summary>A list of proposals with mentioned issues or criteria</summary>

(if not relevant, see first edits in mentioned issues)
* #58 - plugin-candidate.
* #55 - just an idea, proposal template disregarded.
* #63 - the author is not sure whether it could be made as plugin or could be part of the core, avoiding presenting more ideas + template ignored.
* #65 - nice-to-have, possibly plugin-candidate, met with disapproval which could otherwise be welcomed.

</details>

**Comment by Calinou** [comment]
Having a third repository makes it even more difficult to manage labels and permissions consistently, so I'm not sure about it. We'd also run into the issue of having people submit proposals and ideas to the wrong repositories.

**Comment by golddotasksquestions** [comment]
> Having a third repository makes it even more difficult to manage labels and permissions consistently

I don't think there is a need to label ideas at all.

> We'd also run into the issue of having people submit proposals and ideas to the wrong repositories.

This is an issue present in all repositories. 
On the contrary, if we had an official `godot-ideas`, we could have a place to send people to, if their proposals don't meet the quality requirements of `godot-proposals`. 
`godot-ideas` then would be a place to flesh those ideas out, to have a place of discussion and maybe, eventually to then formulate a GIP. in `godot-ideas`, issues can be left open infinitely, until issue owner closes them.
Maintenance would be much lower, because there are less criteria to meet in order to submit an idea. If it is a shit idea, people will ignore it. If it is a worthwhile idea, it might spark a discussion. The likelihood of participation or interaction with other users would be the same as they have been until now when submitting proposals: very low. The tools to increase the likelihood for interaction on that idea are the same as they have been as well: Post the issue on Reddit, Twitter, Discord ect ...
Even if users don't post anywhere, they have documented their need and secured a place for discussion. They can refer people to that place.
Everyone who submits an idea has to have a Github account and come to https://github.com/godotengine: One click away from submitting a bug report or contributing to discussions. It's a low barrier of entry to the become eventually become a meaningful contribution unlike `godot-proposals`.
Regular users, less experienced users would have an official place to share and discuss ideas, small improvements of existing features, or missing features that are the reason they have not started a project in godot yet.
Contributers who want to do actual work can stick to `bugs `or `godot-proposals`.
If contributers or people who want to be inspired are looking for brain farts, maybe good ideas by users who have not fully understood the system yet, discussions on ideas not ready for GIPs, small life improvements that don't justify writing up a ull GIP, random expression of needs from the general userbase, they can go to godot-ideas.

Advantages compared to a platform like Reddit:

- Reddit is all about visibility and has no attention span. Github issues can stay relevant over years. (there are many issues discussed over a period of 4 years on Github) 
-  If someone comes back to a Reddit post it by any chance, they won't even upvote (because why? upvote on Reddit only have a influence in the first a couple of days  of a post), they won't comment, (because people hardly ever reply on old theards). Even a six month old Reddit post is completely pointless for a discussion. On Github issues can get "alive" any day if there is the need.
- The Thumbs up/down on Github is to express support. The up/down on Reddit is to influence visibility. (-> Reddit works as social media platform, to spread ideas, not so much to document ideas)
- On Github thumbs up/down is not anonymous. If someone disagrees, It's easier to find out why.
- Github issues are easier to refer to if you submit a bug or an actual proposal.

**Comment by Xrayez** [comment]
> I don't think there is a need to label ideas at all.

I was thinking of having `core`/`module`/`plugin` labels to potentially filter those that could pass `godot-proposals` requirements, but then again it doesn't have to be like you have to put these lables on each of the created issues unless someone from core contributors express interest.

**Comment by willnationsdev** [comment]
I too believe that this would be a good idea. The balance we need is a low-maintenance, official (and therefore, ideally GitHub-hosted) space in which to spark discussions on potential engine/module/plugin/asset features. If, for example, someone opens a Proposal and they think it will be something that is needed to do in-engine, and then a contributor tells them that an EditorPlugin can do what they are requesting, then the Issue gets closed (potentially) and the case is closed. But the OP is still going to want a space in which to track the idea itself, discuss implementation with the community, etc. And the OP may not even be capable of implementing the idea they have, so they can't just keep it offline to themselves and one day find fellow interested users who DO have the ability to make it.

**Comment by willnationsdev** [comment]
I've actually discussed this as an Issue before since there is not official place to discuss plans or designs around creating *plugins* for Godot (like blurrymind's Dragonbones integration or event stack visual scripting language concept). Previously, when someone opened an Issue in godotengine/godot with an idea for something, and that idea wasn't suitable for the engine, they would be met with tons of negative feedback about why they were even posting the idea, that it doesn't belong, etc. And the contributors are right in the sense that godotengine/godot really shouldn't be the place for that, but the problem is that these kinds of things just don't have a space right now. And until there is one, you'll likely have a lot of frustrated content creators/users out there. That's where a lot of the negative feedback about the new GIP system is coming from, afaict.

**Comment by Xrayez** [comment]
> they would be met with tons of negative feedback about why they were even posting the idea

This actually explains why people tend to bypass discussing ideas in the first place and go straight towards implementation/PR so they can't be as easily rejected. Having a more welcoming place to discuss stuff would save a lot of unnecessary work in case the implementation is a plugin candidate.

**Comment by slapin** [comment]
I think we also need repos where ideas will be buried and ignored by everyone.

Jokes aside, currently almost all of contribution ideas go to "make gdnative" which is equivalent to "get lost". Because gdnative is bastard child and is quite unusable. It is much better to use your custom engine fork than gdnative, because at least your engine change will likely work on all platforms as long as you can compile the engine. Also it is much easier to use, no platform-specific files hanging around, no trouble building, everything was done for you and tested. As developers don't care much about gdnative, I consider the direction as "we don't want anything not made by us, do your own fork" motto.

**Comment by Xrayez** [comment]
Perhaps the repository shouldn't be named as `godot-ideas`, but actually `godot-discussions` which would serve a different purpose from actual proposals or a random collection of ideas (though nobody prevents you from presenting an idea in a proposal form rather than discussion).

You might say that forums could be used for discussions. But as stated by @golddotasksquestions, GitHub has a lot more benefits for this, being able to cross-reference issues is my personal favorite 😛.

godotengine/godot#19486:

> Sorry for the very long post. I'm not mad if you don't want to read the whole thing, so here's a short version of it.

People tend to be guilty starting discussions, and no wonder.

[Trackers](https://github.com/godotengine/godot/issues?q=is%3Aopen+is%3Aissue+label%3Atracker) or TODO lists could also be created to gather related ideas and discussions together (notice these kind of issues would be forbidden with the current GIP process).

**Comment by Sslaxx** [comment]
> Jokes aside, currently almost all of contribution ideas go to "make gdnative" which is equivalent to "get lost". Because gdnative is bastard child and is quite unusable. It is much better to use your custom engine fork than gdnative, because at least your engine change will likely work on all platforms as long as you can compile the engine. Also it is much easier to use, no platform-specific files hanging around, no trouble building, everything was done for you and tested. As developers don't care much about gdnative, I consider the direction as "we don't want anything not made by us, do your own fork" motto.

If GDNative is a "bastard child" and "quite unusable", then it's better to file bug reports about it than to complain. Tell the devs *why* and *how*. At the moment the problem with GDNative is its main developer has gone on to other things, so it needs someone who can take it over and work on it. I'm not sure it's fair to say that "developers don't care much about gdnative", though.

**Comment by golddotasksquestions** [comment]
> Perhaps the repository shouldn't be named as `godot-ideas`, but actually `godot-discussions` which would serve a different purpose from actual proposals or a random collection of ideas (though nobody prevents you from presenting an idea in a proposal form rather than discussion).

I would feel less inclined to post under `godot-discussions`. This sounds like work. And honestly a little discouraging if 90% of all issues in `godot-discussions` don't actually have a discussion.

Posing an idea on `godot-ideas` is just that. You post it and you are done. There may evolve a discussion around it naturally, maybe people rally around it and want it to be a GIP, but an idea is still a documented idea even if no one reacts to it. 
If someone else later posts a very similar concept or idea the original poster can refer them to your issue (duplicate). Correlations and differences can be discussed. By placing an idea on the official Godot Github which contains **THE SOURCE**, it creates a minuscule sense of ownership. I'd already feel like as I am part of the project.

**Comment by Xrayez** [comment]
I made this "meme" and posted on Reddit, it might not be appreciated there as much, so posting here as well (please don't feel offended) :)

![godot-bug-or-feature](https://user-images.githubusercontent.com/17108460/64538331-1de04b00-d325-11e9-966a-ec4f26d512ee.png)

**Comment by Xrayez** [comment]
I've updated/restructured/added some of the ideas in OP, so you know [edited]. 🙂
Especially the human factor part.

**Comment by lawnjelly** [comment]
While I really like the idea of the proposals system and see it as a big improvement, I think there is a second great opportunity here. As mentioned earlier in this thread I think there is currently a big void between features being developed for core, and those being made as addons / gdnative / modules etc.

The whole philosophy of Godot as far as I can see is to keep the core 'lean and mean' (if you'll excuse the windows analogy), and move as much of the functionality as possible into optional components (interpolated camera, *cough*). If that is the aim then we should surely try and move a little more towards treating the development of non-core features as first class citizens. I.e. if you make a proposal and it is deemed that it is better to be non-core, this should not be seen as a negative. After all, it should be seen as a success, that the core has been built in such a way as to make this possible.

What would be great (although I don't know if possible) would be to have a parallel system for core proposals and non-core, and if something is deemed non-core, then the topic can be moved between the two (or vice versa), so that any discussion etc is not lost.

Along with that I think there is potential for the more techy guys to streamline the solution for the rest of us for building GDnative and perhaps modules for different platforms. It should ideally be as easy to get something available via non-core as it is through core (or, even easier because the review process can be less stringent).

Speaking as a developer, just as there is continuous integration scheme for the master branch, it would be nice to have an official CI for gdnative and perhaps modules too. So developers would periodically upload their latest stable version, and the official CI would churn out binaries, that any user could then download (from the IDE) and add to their local godot installation. Essentially most developers are more interested in programming than in running a bunch of builds for different platforms and faffing with distribution logistics. Well I know I am! :smile:

**Comment by slapin** [comment]
Can't disagree, but moving features out of core is not architectural
decision, it is just resource management. So it is signal to "move out of
our sandbox"
and go elsewhere do your thing. So no, currently getting into core
feature-wise is very unlikely, so you have every opportunity do do
everything on your
own, which is out of sight, out of mind business. As everything which can't
be done out of core is "not needed in all games" too. So currently for any
contributor the best advice is not waste time but do your own engine fork.
You can be kind and make others easy to pool from you and make feature
branches. All bug fixes are generally upstreamed though.
So all features we can get currently are either made ourselves or by core
developers, we can't contribute features.

**Comment by willnationsdev** [comment]
@slapin I'm not really sure what you mean by your comments? People can contribute features. They do it to the engine's core (`/core`), the core engine (`/core`, `/main`, `/scene`, `/server`), as well as the main engine repository (versus external repos like addons, plugins, and modules).

Yes, I would agree that the fastest way for someone to do whatever they want with their code is to add it to their own fork of the engine (there's no approval process that way), but this Issue isn't targeted at the audience of people who only care about making their own stuff. It's about people who want to contribute features to the publicly accessible engine either by 1) adding a feature to the main engine repo or by 2) creating their own public repository with an addon/plugin/module. And the problem is that there is no official Godot tracker (like a GitHub repo with Issues) for keeping track of the second set of proposed changes (for people who make addon/plugin/module proposals but don't have the time/knowledge/experience/motivation to work on them - yet).

This Issue is about establishing an official one, so, unless I'm mistaken in understanding what you said, I don't think you're comments were quite on topic.

**Comment by golddotasksquestions** [comment]
> but this Issue isn't targeted at the audience of people who only care about making their own stuff. 
Or **can** make their own stuff. Many incredibly valuable improvement ideas, especially in the UI/UX department come from people who **use** the engine, but don't necessarily know how it works behind the curtains. Often these people even have a better idea what the software should do, because their perception is not tainted at all by the reasons why things are done one way or another.

**Comment by girng** [comment]
This is why I've always advocated for at least one centralized location.  For example, the forum in 2014 or so (can't remember the exact date), but you'd instantly check it in the morning, or if you just randomly think about Godot / game dev stuff. It was the **go-to** place. Your topic could be seen not only by core developers, but contributors, fellow game devs, etc.  It had a lot of different sections (scripts, tutorials, plugins). No upvote/downvote system, just a community where your Godot related ideas/topics could be seen  and have an opportunity to flourish.

That was a really cool  because it was like the hub for Godot.   People can still do this now through the community channels (discord, godot dev forum, irc, etc, etc, etc). However, IMHO having that one "central hub" forum was ideal.  

I honestly believe more separating at this point is just not a good idea, we need more centralization! (if that's a word). That's why I was super excited to hear about the godot-proposals repo, and happy to see reduz posting about it on twitter / news announcement.

**Comment by willnationsdev** [comment]
@girng except that godot-proposals is specifically designed to filter out many of the ideas that would have been permitted to exist in something like Godot forums. That's kinda the point of this proposal.

**Comment by girng** [comment]
@willnationsdev Yeah, you are correct I misspoke at the end. I mean in the context of   godot-ideas/godot-discussions

**Comment by Xrayez** [comment]
> I honestly believe more separating at this point is just not a good idea

I don't want that either tbh, but seeing how restrictive the current process is now and my failed attempts to waive some of the restrictions in #39 I hardly see a better alternative (this proposal is made in a way to let you accept the other alternative in fact).

So for now the only type of proposals I'd make are in fact those which would allow to introduce independent changes to Godot more easily (better module linking support, seamless GDNative infrastructure without having to write `Sconstruct` on your own etc). 

I learned that my needs are too specific to be even discussed.

At the very least, these `plugin-candidate` proposals should not be bashed (or not as hard).

> is no official place to discuss plans or designs around creating plugins for Godot

So yeah, I'd change the title to this quote, but there are still ideas which are impossible to implement via plugin system and which could still be controversial.

**Comment by golddotasksquestions** [comment]
> I learned that my needs are too specific to be even discussed.

I think that's a very problematic situation for the development of any editor, especially if it is open source.

1. because it's impossible to tell if your needs are in fact too specific before you put them out in the public and they had enough time and occasion to resonate with other users
2. because others might think the same about their issue, not daring to speak up with the result that everyone thinks it's just their specific problem
3. because the fewer people participate in total, the less accurate will be the assessment of any need in the community of users even if you have a fair bit of participation on some issue.
3. because there is actually a good portion of luck involved to have people with the same issue meet at a common place. I online search for issues all the time and still stumble across really old ones from years ago I after months of having the issue just because someone just posted a direct link somewhere.

So my point is we should encourage people to come out of their holes and communicate more, share their issues more. For this to happen we have to make sure every issue a user has, every concern is treated as relevant and important. The challenge should not be to prohibit the motivation to participate, but rather to channel the motivation users have to the right places.

**Comment by amisner2k** [comment]
So are we doing this? What's the holdup. Who's the person who decides that this repo gets created or not or does somebody just have to do it? Who has the keys to this? Let's make it happen.

**Comment by willnationsdev** [comment]
@amisner2k it would have to be the godotengine organization admins who would have the GitHub authority to create new repositories affiliated with the organization (to get, say, a godotengine/godot-ideas URL). And it will probably have to wait for Akien and/or reduz to make a final call on it, perhaps after discussing on IRC with the other core devs.

**Comment by amisner2k** [comment]
Cool cool......well if anything. Shouldn't hurt to create it and see how it goes. If for some reason it doesn't pan out it can always be deleted in the future. I just hate to see things die slow deaths with endless discussion.  But yea, hopefully they see this and make it happen.  I'm rootin' for ya, Will.  :)

**Comment by clayjohn** [comment]
Keep in mind, before further steps are taken, there would have to be consensus on the issue. 

Additionally, for this issue - in my opinion - there also needs to be more concrete decisions on how the alternative repo gets managed and moderated. Currently a handful of people put in an incredible amount of work to barely manage the current issues (Keep in mind we are currently failing). So any new system on top of the current one needs to reduce the workload of contributors not add to it. In my opinion that hurdle needs to be solved before creating a new repo. Not after.

**Comment by amisner2k** [comment]
Everything I've heard so far about this is that it wouldn't require much if any workload of contributors to "manage". It also seems to be designed to reduce the number of issues in the main repo so that they can instead be dumped in this "ideas" repo. That's my understanding anyway, feel free to correct me if I'm wrong.

**Comment by clayjohn** [comment]
@amisner2k We can't just invite the public onto a platform hosted by us and then just throw up our hands and avoid it. First of all, the content has to be moderated for spam, hate content, etc. Second, in order to be used, someone has to determine which issues are worth opening a proposal over. Third, very quickly we will reach thousands of suggestions, at that point it will be nearly impossible for users to find duplicates, accordingly many more duplicates arise which will inhibit productive discussion. This third problem will continue growing indefinitely (as it is now in the main repo).

Just asserting that something won't add to the workload of contributors doesn't make it so. Many users also think that maintaining the current issue list is low-effort when in fact many people spend many hours a day just to keep it at its current level of organization.

Lastly, and I know this doesn't go to your point. But I still don't understand the benefit of having this in a repository. Godot isn't suffering from a lack of half-baked ideas right now, in fact, we are drowning in them. We have no clear way of distinguishing good from bad ideas right now. Godot-proposals solves that problem, it cuts off half-baked ideas at the source. This proposal seeks to bring back all the low-quality proposals for same ill-defined gain.

**Comment by amisner2k** [comment]
Why do we have to monitor everything for spam and hate content. Who says? This is an open-source, open community. I see your point.

It's better to have thousands of suggestions in a "ideas" repo then in the official repo...because that will happen in the future if Godot's popularity continues to grow.

Just asserting that something *will* add to the workload doesn't make it so. I don't mean to be confrontational or anything just saying.

Let's just go for it. There's no need for all this bureaucracy. All I'm saying is that others here have brought up some pretty valid concerns and I feel like it's worth trying.

**Comment by clayjohn** [comment]
> It's better to have thousands of suggestions in a "ideas" repo then in the "core" repo...because that will happen in the future if Godot's popularity continues to grow.

I agree that would be better. However, having no ideas in the "core" repo and a smaller set of manageable and well thought ideas in Godot-Proposals is even better than both. 

> Just asserting that something will add to the workload doesn't make it so. I don't mean to be confrontational or anything just saying.

Good thing I'm not *just* saying then. :) I'm speaking from experience.

> Let's just go for it. There's no need for all this bureaucracy.

Very easy to say when the work doesn't fall on your lap. Much harder to agree with when your the one that will have to deal with it. 

To clarify, I am not against this suggestion. golddotasksquestions and I had a long conversation about it when it was first opened and he made some very convincing points. I just think before we take the next step we need to address how it will be implemented in a way that minimizes the work required from contributors. Keep in mind, we contribute because we enjoy working on the engine. We moderate the github repo because someone has to. The more work it is to moderate, the less we get to work on the engine, the less reason we have to stick around. It isn't fun to moderate the github repo its work that we don't get paid for and we get nothing out of trying to appease the 1000s of people who have half-baked ideas.

**Comment by amisner2k** [comment]
> manageable and well thought ideas in Godot-Proposals is even better
Completely agree. If I understand what others here have suggested, we can think of the "ideas" repo as a sort of "proving ground" and a way to hash out ideas that can then be promoted and cherry picked to get made into proper full-on proposals which can then go into the Godot-Proposals repo.

> Very easy to say when the work doesn't fall on your lap. Much harder to agree with when your the one that will have to deal with it.
I hear ya, but I mean...how much work would it really be? You're making it sound like people will have to dedicate just as much resources and time to it like the other two Core and Proposals repos, but others here seem to be trying to convey that it would not necessitate much management or effort to maintain.

> We moderate the github repo because someone has to.
Since I'm clearly new here (as you can well and surely tell), what exactly happens if you don't moderate a github repo here? Just curious. Because in my utmost ignorance, my understanding would be that the work would largely entail occasionally seeing if anything interesting was added that warranted attention.

The biggest hurdle that seems to be brought up here is that people want to feel like they can bring up a suggestion without worrying about how it's received or fearing that they didn't type up a proposal right. There's an intimidation factor at work. I think if it's made as abundantly clear as possible that the ideas repo is *not* a guarantee of anything and that it's sole purpose is to gather feedback and hash things out before *maybe* making it into a proposal...that will help to alleviate your concerns for maintenance and moderation.

Hope that makes sense.

Either way.....I will be ok with whatever happens.....it seems like what's at the root here is breaking down barriers to entry and making it easier for newcomers to feel like they're contributing in at least some small way. I feel like the proposals repo will just end up getting cluttered with an elevated level of submittals in the future. The "Proposals" Repo is strict (for good reason), the "Ideas" Repo is posited as being free and no-holds barred...precisely the right environment for ideas to take shape.

There will be a lot of bad ideas...no way around that unfortunately.  :P

**Comment by amisner2k** [comment]
Hmm...you know the more I think about it...the more I realize that a new repo *would* add more maintenance burden for contributors without enough benefit to new users. I suppose if we're to do this at all we ought to do it right. We need to keep in mind that as Godot grows, there will be more and more users coming in whom have little to no intention on contributing code and instead just want to voice some feedback and therefore, perhaps GitHub is not the place they should go to provide feedback.

Now that (https://godotforums.org/) is back and officially a thing (and actually looks quite inviting), perhaps this "Ideas" thing should live there?

Thank you @clayjohn for being very patient with me and helping me understand the issues at hand here.

**Comment by girng** [comment]
I think we all just want to be connected in some form or another (in the context of Godot :P). And most importantly, have our thoughts and ideas be heard.  IMO,  communication is key, and it will make the community flourish  not only as individuals, _but in the creation of suggestions that further improve Godot_.  That's powerful and I hope we never lose sight of this!

**Comment by golddotasksquestions** [comment]
@clayohn
> First of all, the content has to be moderated for spam, hate content
Yes. As much as any content on the internet. It's the inevitable downside of being open and accessible. Do you have any data on how spam and outright hate you guys have to sort out already?

>Second, in order to be used, someone has to determine which issues are worth opening a proposal over.

I get the impression we are not talking about the same thing here. This is not a place where I would Core contributors to hang around. "Ideas" are not something that is a blueprint for a design, a design someone just has to sign off and then it gets made. If an idea is that far along, they can post in "Proposals" as the outline guidelines states when you want to submit an issue and on the Repository home page as well.
No one has to come to a person and tell them, "This is a good, solid fleshed out idea, I think you should submit this as proposal!". If someone really thinks so and wants to they can, just like it already happens on Discord and Reddit. There is no need for core contributors to be involved in this process.
The problem with those discussions on Discord and Reddit is that they are fleeting. There is no record of it, no meaningful documentation or ongoing exchange over a long timespan. The fact that we need to have a place for these discussions is already abundantly proven by countless "ideas" and "proposals" which are already discussed since 2016 or even longer.

> Third, very quickly we will reach thousands of suggestions, at that point it will be nearly impossible for users to find duplicates, accordingly many more duplicates arise which will inhibit productive discussion.

Why would you _want_ to find duplicates in "Ideas"? If someone there engages in a discussion and does, I'm sure it's welcomed. But unlike "Proposals" or "Bugs", it is not necessary to find duplicates of ideas. Besides, people who submit in Ideas on Github can use the the excellent auto and manual search, if they do want to find duplicates.

> Godot isn't suffering from a lack of half-baked ideas right now, in fact, we are drowning in them. 

That's why we would would like to tunnel those half-baked ideas to Godot-ideas along with other stuff. Simply outlawing bad ideas is like trying to disinfect drinking water with radioactive waste. Sure it works, but it's probably not the healthiest of all options.

>Many users also think that maintaining the current issue list is low-effort when in fact many people spend many hours a day just to keep it at its current level of organization.

Anyone who sticks around for a while can see the work that goes into. If a user just pops in and out once to drop their demands and want's it done immediately, without actually caring about the requirements or process for a PR or GIP, we can gently and friendly escort them to Godot ideas, where they can do their unruly business.

> I still don't understand the benefit of having this in a repository. Godot isn't suffering from a lack of half-baked ideas right now, in fact, we are drowning in them.

The benefit, the purpose would be 

- to foster participation, involvement, 
- to serve as gate, 
- to lower the barrier of entry, 
- to migrate lower quality proposals that won't fulfill the criteria of a blueprint, 
- to give a place to people who are unsure it their issue is a bug or a feature request, 
- to document demand and interest on far far away features ... 
- to recruit people who show an interest in doing the aforementioned moderated for spam and hate.

**Comment by clayjohn** [comment]
> Do you have any data on how much spam and outright hate you guys have to sort out regulary?

Fortunately not a lot. While we regularly deal with this sort of thing it isn't frequent. But we aren't currently hosting a place on the internet where we say "come say whatever you want without moderation"

> I get the impression we are not talking about the same thing here. This is not a place where I would expect to see Core contributors to hang around...

I understand that. But I am certain that if we create a space for people to discuss idea, their expectations will be that they will be heard. Currently, we make no such warranty and yet we have people asking questions in the bug report template demanding that they be addressed by the core devs directly. If we create a space and attach the Godot name to it, some amount of people will expect that the core devs interact with them there. By connecting it to the main Github repo, we will be creating an expectation in the minds of users. 

I know that many of you will be able to use it without asking anything of the core devs. But I am also certain that many people won't see if the same way and we will get comments about them being ignored and also "why would you open this repo at all if your just going to ignore all my great ideas?"

>  If a user just pops in and out once to drop their demands and want's it done immediately, without actually caring about the requirements or process for a PR or GIP, we can gently and friendly escort them to Godot ideas, where they can do their unruly business.

My point is exactly that *someone* needs to do this gentle escorting and it is going to fall on contributors, as we are doing now. 

> The benefit, the purpose would be ...

I mostly agree. :) Again, I am not against this proposal. I just think we need to figure out how we are going to implement it without increasing the moderation burden on contributors.

**Comment by willnationsdev** [comment]
@clayjohn 

> By connecting it to the main Github repo, we will be creating an expectation in the minds of users.

I can see the dilemma there. On the one hand, we want it to be recognizable as the "main" channel for people to go to for discussing ideas related to Godot, a place to point people to, but keeping it in the godotengine organization would also invite expectations of involvement from the core contributors. Perhaps the better solution then is to, through mass participation, create an unofficial ideas repo that is still located on GitHub (making it easier to reference Issues and PRs in other repositories), but that specifically ISN'T invited to be a place that core contributors are expected to participate in or moderate. That would answer a lot of your concerns in regards to preventing the growth of contributors' burden. The biggest point against doing so it that it makes it harder to point to it as the go-to place for ideas, but I think that could in large part be solved by updating Issue templates to refer people to "unofficial channels" for discussion, linking the ideas repo.

I am imagining something that would function more along the lines of GitHub's [isaacs/github repository](https://github.com/isaacs/github). It is an unofficial place to submit issues for GitHub, and has no expectation that GitHub team members will participate, but you can bet your butt that it's the prime location for the community to submit ideas for improving the software.

> My point is exactly that someone needs to do this gentle escorting and it is going to fall on contributors, as we are doing now.

If we took this approach, it would be those who are running the unofficial repo (and the volunteers who moderate it) who handle this, not people associated with the engine directly. It would function more like a forum and be more easily accessible since you don't have to be a member of the godotengine organization to be involved as a moderator.

**Comment by golddotasksquestions** [comment]
> But we aren't currently hosting a place on the internet where we say "come say whatever you want without moderation"

Well as amisner2k said, we can try and if it gets completely overrun by trolls, readjust or close.

> "why would you open this repo at all if your just going to ignore all my great ideas?"

Because this community does not only consist of core devs. Ideas would be a place with far less participation of core devs, but also less requirements to post. Proposals and Issues are repos with more participation of core devs, but also have higher requirements for a post. I think it's really simple and very fair principle. Also simple to communicate. It is important to manage expectation.

> Currently, we make no such warranty and yet we have people asking questions in the bug report template demanding that they be addressed by the core devs directly

You cannot take people who seriously think they can make demands in an open source project seriously. For your own sanity, I really hope you don't. I'm not a contributing core dev, but if I were, this this person and their issue would be immediately on my ignore-list. Providing them a link that explains FOSS is almost too generous. Maybe if I had a good day.

> If we create a space and attach the Godot name to it, some amount of people will expect that the core devs interact with them there. 
Just spell it out that that won't happen then. The pro and con for posting in either Repo has to be obvious: In "Proposals" the user has a better change to get a core-devs attention, but also has to submit to stricter and more specific formalization criteria and submission rules. In "Ideas" there is less formality, but probably also a different, less experienced crowd. 

> By connecting it to the main Github repo, we will be creating an expectation in the minds of users.
For those who participate, being a ever so tiny but still relevant part of something huge and amazing, I would hope.
Any other expectation can be managed by clearly outlining the purpose and use of the repo.

> My point is exactly that someone needs to do this gentle escorting and it is going to fall on contributors, as we are doing now.
I was drawing a mental image. What I meant was to just remove the issue from the repo and attach it to another. (Not sure how this process works on Github, but it seems to work because it was mentioned about the all the valuable enhancement proposals of the 5000 open issues in main)
So yes, someone has to do that. But how is that different from closing an issue (which usually involves a lengths discussion?) "Reparenting" the issue from "Proposals" or "Bugs" to "Ideas" would not close it, it just says you are having this talk at the wrong place, not that is not welcome. 

> Again, I am not against this proposal. I just think we need to figure out how we are going to implement it without increasing the moderation burden on contributors.
Totally agree!

**Comment by amisner2k** [comment]
@golddotasksquestions  What do you see as the biggest reason not to have this sort of thing on (https://godotforums.org/) ?

**Comment by Calinou** [comment]
I'm still not sure about the idea of having an ideas repository. Many proprietary software developers have idea trackers set up, and they often end up being like a `/dev/null` equivalent. If core contributors aren't going to engage in it (it's harder than it sounds like), it'll likely end up being a waste of time for everyone.

Funnily enough, while looking for examples of idea trackers used by proprietary software developers, many have been deprecated recently or replaced by a "second class" version like [feedback.discordapp.com](https://feedback.discordapp.com/) which now just redirects to a forum section.

**Comment by amisner2k** [comment]
@Calinou  does bring up a very fair point. Still...I'm interested to hear someone chime in on putting this sort of thing as one of the sections on (https://godotforums.org/)

**Comment by willnationsdev** [comment]
@amisner2k I think the biggest perceived obstacle for using the forums is that, if it isn't on GitHub, then you don't have built-in reference tracking between Issues in different repositories. On the other hand though, if the intent is to not distract devs as much as possible while discussing these ideas, then backward references that pop up in actual GitHub issues may be something you wouldn't want anyway. In which case, the forums would be a good place to discuss things that doesn't bother people on GitHub at all, but still gives people a place to discuss things. The forums have their own moderators after all, right?

**Comment by golddotasksquestions** [comment]
@amisner2k 

> What do you see as the biggest reason not to have this sort of thing on (https://godotforums.org/) ?


I would want more people from a wider group of Godot users (not just coders) to have GitHub account, so they can submit bug in their specialty field and together build a better representation of Godot total user when it comes to feature discussion, usability, whether or not something is needed as well as the process of finding problem solution. Different minds offer more perspectives. It's going to be hard to bring artists and musicians to a platform as is. But Github is the place were these decisions are discussed and made.

**Comment by willnationsdev** [comment]
@golddotasksquestions 

> I would want more people from a wider group of Godot users (not just coders) to have GitHub account

But if users don't need to have a GitHub account (a programmer-favored platform) and instead only need a Godot forums account (an open platform for all Godot users), it would be easier for ideas to be discussed by exactly those people. If our equally primary goal is to make sure the space doesn't negatively impact the existing GitHub users, then the forum would really be the most optimal place to put it. It's still a Godot-flavored space, but it has a separate team moderating it (presumably could be delegated further?) and doesn't have the same focus from the core team as GitHub Issues do.

I can see the desire for people to create GitHub accounts for the sake of reporting bugs, but I don't think we should hamstring the ideas discussion space (or hamstring the core team) by forcing it onto the GitHub platform just for that sake.

**Comment by amisner2k** [comment]
Yeah I'm thinking that if someone really wants to get involved contributing code to one of the repos or submitting actual bug reports, they will take the necessary next step to create a GitHub account and go from there. That would be my hope anyway.  :P

**Comment by golddotasksquestions** [comment]
You are right that it would be a good place if we somehow could convince people that Forums are back en vogue. But even if we did (which I find unlikely), the Forum is still not the place to report bugs, it's still not the place where PRs and discussed before they are merged, or feature proposals are submitted. 

What some of you seem to forget is that more people on the Godot Github also means more Core devs in the long run. Channeling is the issue, not that people want to participate.

**Comment by amisner2k** [comment]
> What some of you seem to forget is that more people on the Godot Github also means more Core devs in the long run.

Well...I'm not entirely convinced that more people on GitHub entirely translates to more Core devs in the long run. It may just be the case that it's just a lot more people with GitHub accounts not really doing anything productive, lol.  But I would certainly hope for the former. ;)

**Comment by Xrayez** [comment]
>If core contributors aren't going to engage in it (it's harder than it sounds like), it'll likely end up being a waste of time for everyone involved.

Not everything pertains to core development. Having such a repo would mean that `godot-proposals` would become "more clean" from plugin-candidate proposals. Seeing how Godot is quite modular gives more hope that such a repo won't be abandoned, so there's a win-win situation here.

> Perhaps the better solution then is to, through mass participation, create an unofficial ideas repo that is still located on GitHub (making it easier to reference Issues and PRs in other repositories), but that specifically ISN'T in the godotengine organization and therefore ISN'T invited to be a place that core contributors are expected to participate in or moderate.

I was thinking of creating `godotengine-community` organization on GitHub for that matter, but I don't really want to hi-jack it that way. But then again, the official `godot-proposals` would ideally need to point to unofficial `godot-ideas` in some way, but then you're concerned that these ideas/proposals will end up in wrong place, while in fact this would signify that there's something wrong with the current GIP process if people start to ignore `godot-proposals`.

Regarding forums, I don't visit them much (if at all) and I've hardly stumbled upon forums via search engines, most search results lead me to QA and GitHub.

**Comment by amisner2k** [comment]
> Regarding forums, I don't visit them much (if at all) and I've hardly stumbled upon forums via search engines, most search results lead me to QA and GitHub.

You should visit them more  :)  Improving its ranking in search results could be improved with various SEO optimizations. I'm new to this Godot community and my first impressions of that site are positive. I think it would be an ideal home for people to get official support and feedback for ideas and issues where they can then be turned in to real proposals here in GitHub...but that's just me (insert kermit sipping tea meme here)

**Comment by clayjohn** [comment]
> I was thinking of creating godotengine-community organization on GitHub for that matter, but I don't really want to hi-jack it that way.

@Xrayez This is actually a really good idea and I would encourage you to pursue it. The forums are another example of a Godot resource maintained by the community. Additionally, the idea when forming ``godot-proposals`` was that the community might rise up to meet the demand for a place to discuss preliminary issues (I thought this would take place on reddit/discord, however, this sounds better). Part of the beauty of open-source projects is that the community can meet the community's needs without the core developers having to get involved.

**Comment by Zireael07** [comment]
Godotengine-community organization is a great thing either way. I imagine in addition to the ideas repo that is being floated around, it could also host Godot-next?

**Comment by willnationsdev** [comment]
@Zireael07 I had initially created the Godot Extended Libraries organization targeted at that purpose, but I think repurposing that concept to a more generic organization, not targeted at just plugins, is a good idea. I have yet to move godot-next to GEL yet even though there is a discord for GEL. We should definitely create that organization. We can then start inviting people to the organization and large-scale migrating plugins to that organization.

**Comment by willnationsdev** [comment]
Here's the icon we made for the Godot Extended Libraries organization. Feel free to use it if you want. The idea was to be a spellbook of some sort: recipes for executing magic.

![icon_gel (1)](https://user-images.githubusercontent.com/16217563/65353776-9364da80-dbb3-11e9-80a0-58c54bcafd80.png)

**Comment by Xrayez** [comment]
@willnationsdev I think this makes perfect sense, you've contributed a fair share of stuff to Godot community-wise and you've already set it up, I could simply transfer `godot-ideas` repo there, if you're willing to rename `GEL` to `godotengine-community`, what do you think?

**Comment by willnationsdev** [comment]
@Xrayez Here ya go! https://github.com/godot-community

**Comment by willnationsdev** [comment]
Now, I don't wanna have to be the only "moderator" for the community, so if there is someone who wants to be super extra responsible and not horribly mess with other people's code in the repo.....now's the time to volunteer. lol

**Comment by Xrayez** [comment]
> Followed the godot- convention

Might be better as it's non-official in fact.

Transferred just fine I think: https://github.com/godot-community/godot-ideas

**Comment by willnationsdev** [comment]
Given all this, I think that if we update the `godot-ideas` proposal template and share the `godot-community` and `godot-ideas` stuff around social media, we should be able to gather new interest in the space. I'll put together a post for Reddit and the FB group.

I think this Proposal is more or less closeable at this point.

**Comment by Xrayez** [comment]
Yeah, lets see how it goes though so that maybe some `godot-proposals` issues can be possibly revealed here, but I'm fine if anyone closes this at this point.

**Comment by clayjohn** [comment]
Also, if successful we can link to it into the community channels doc and on the readme here. :)

**Comment by willnationsdev** [comment]
Here's the Reddit post: https://www.reddit.com/r/godot/comments/d71adf/github_organization_godotcommunity_1_new_home_for/

**Comment by amisner2k** [comment]
This is amazing you guys! I love the solution you all came up with. @willnationsdev , I'd be willing to help moderate if you'd like, just tell me what I need to do. I can spare a few spare moments to help out. I love what you guys are doing here.

**Comment by willnationsdev** [comment]
@Calinou Is there any hope of creating a reference to `godot-ideas`' existence in the issues template(s) for godot and/or godot-proposals, in addition to godot referring people to godot-proposals? That might help keep people from falling through cracks. Not sure if that's desired though as it may also lend credence to some association with the godotengine organization. I feel though that if unacceptable Issues were to be migrated wholesale to godot-ideas as appropriate, then that kind of collaboration might engender some long-term relationship worthy of that. Figure I'd bring it up for discussion anyway.

**Comment by willnationsdev** [comment]
In the interest of avoiding any confusion with official godotengine channels, I have renamed the organization back to `godot-extended-libraries` and will leave it at that for now. We need to let @akien-mga and @reduz take a look at this Issue before we really greenlight things (a little late for that though - whoops).

**Comment by Xrayez** [comment]
Updated `godot-ideas` description to better convey the actual intention, the repository was meant to demonstrate an alternative to `godot-proposals` but the ideas have diverged greatly.

As for renaming, "community" stands for "users" and "engine" mostly stands for "developers", so I'm not sure what other name could be chosen for this...

@akien-mga and @reduz should really make the final call then, but there's clearly an issue for me here (and some others here), the reason why I started to discuss this early is to prevent users to become discouraged with the process, so I have the best intention, sorry if I've created any impression of hostility here, peace.

**Comment by reduz** [comment]
@Xrayez I think it's too soon to do anything, and we haven't even tested the new system because @akien-mga was on vacation. We will have enough time to see how it works and how we can adjust it over time so most can be satisfied, so I feel that all this is an overreaction at the moment.

If you want to do your own system in parallel feel free to do it, but it won't be official, nor it will need official approval.

**Comment by reduz** [comment]
The hard line of conflict will probably always be the dismissal of proposals that don't have user interest, or any practical use in existing projects (even if they sound cool), and this won't change because we want to avoid bloating Godot with stuff no one will use.

That said, and as others mentioned, we are a lot more open on feedback for improving external plugin support, including the plugin API and GDNative C++ so extending as outside plugins is easier (it's kind of a hassle for a lot of things). You feedback for these kind of features that are on a gray area will be a lot more interesting if it's about improving plugin API development so they can actually be implemented externally.

**Comment by Xrayez** [comment]
@reduz my overreaction is mandated by the [survivorship bias](https://en.wikipedia.org/wiki/Survivorship_bias) which could be quite real here. People who want to make a proposal which doesn't match up to the standard/requirements could just decide to close the tab because most don't have any authority nor influence to suggest an alternative. And those who do notice potential problem would remain in minor group, so one could make a logical error that the issue is not really there nor it is objective enough to allow any changes to be made.

I think discussing this won't hurt anything. No actions are needed for now indeed, let the system be tested, I go idle mode. Sorry for making it appear too serious of a problem. 🙂

**Comment by reduz** [comment]
@Xrayez Yes, but then again, all you have to do is explain how it would be of use to you in a real-life project. I think it's a fair requirement.

**Comment by reduz** [comment]
Your only "risk" is basically a situation where:

* You had a great idea
* Can't really find an use for it, or have no idea whether it's going to be useful or not
* So, if someone does, it would be for reasons you were not sure about, or did not expect.

This is a too corner case situation, and a terrible criteria for adding new features.

**Comment by willnationsdev** [comment]
@reduz 

> This is a too corner case situation, and a terrible criteria for adding new features.

I think we all agree that godotengine/godot and godotengine/godot-proposals shouldn't be cluttered with such Issues.

> It's just not worth it.

I can personally see how devoting manpower to Issues with such vague usefulness could be viewed this way. It *might* lead to like-minded people finding each other. They *might* discuss it in depth. They *might* actually build something out of the idea. It *might* end up solving *some kind of problem* in the future. It's all so nebulous.

> It just works much better the other way, when you find a real-life obstacle and can describe it, then you or the the community can work together and think of ways to overcome it.
> Our mission is about making something useful for others based on their real needs, not having fun adding features that we find cool, that no one currenty needs, so that may nor may not be useful.

I agree wholeheartedly with these statements. Which is why I *still think we have a gaping hole* in our Issues at the moment.

In particular, we do not have a proper tracker for issues where the submitter...

1. Knows a use case for an idea.
1. Has met with obstacles because their idea doesn't exist yet.
1. Knows the idea itself *could* exist outside of the engine itself.

Examples include...

1. an addon with assets.
1. an EditorPlugin.
1. an engine module.
1. a template project.
1. a full project that provides a useful tool for the community.

These entire category of ideas don't belong in godotengine/godot because they aren't bug reports and don't belong in godotengine/godot-proposals because they aren't necessarily feature requests or enhancements for the core engine repository. Having a GitHub repo allows us to fully transfer away all issues in /godot that are more suitable for addons (like the Dragonbones plugin stuff that was recently closed).

A good example is #13. It would be convenient to have it built into the engine, sure. But we decide it shouldn't go in the engine, then I suddenly have *no prescribed place on GitHub* to track the idea. This is where `godot-extended-libraries/godot-ideas` could be useful. I already made the GEL organization a ways back to hold community plugins. We've got `godot-tensorflow`, `godot-next` and the `godot-plugin-refresher` in there (latter two are mine). This is especially useful to me since I can now fork those repos and keep all of my experimental WIP feature branches to myself rather than in the repos that the community largely looks at.

The godotengine organization *could* host something of that sort, but @Calinou suggests otherwise:

> Having a third repository makes it even more difficult to manage labels and permissions consistently, so I'm not sure about it. We'd also run into the issue of having people submit proposals and ideas to the wrong repositories.

As such, I thought it better to have a separate organization that doesn't have that expectation and which maintains a separate "ideas" repo especially for tracking such things.

**Comment by Xrayez** [comment]
## To summarize

The expectation that the new GIP would be simply a separate place for discussing and proposing features is not met (at least for me). Instead, a new set of restrictions is added with an intent that the place doesn't become cluttered, promote high-quality proposals which are easier to maintain and review by the Godot Engine members (for a lot this is also a volunteer/unpaid work).

Based on the feedback so far, I think it's very unlikely for something akin to [`godot-ideas`](https://github.com/godot-extended-libraries/godot-ideas) to be officially maintained by Godot Engine members that way. Some (failed) attempts were made to modify the existing proposal system to be more flexible as in #39, #42.

We see how some people still ignore the proposal template, or fill out questions just because they're required to fill, so the quality of most current proposals is still quite low.

Having said that, I think it would be good if users could choose whether they just want to share an idea which is "nice-to-have" and propose a feature which is "need-to-have", so please consider adding a link to unofficial [`godot-ideas`](https://github.com/godot-extended-libraries/godot-ideas) repo: #91.

The system needs more testing, the discussion is quite controversial and this particular meta proposal can be closed for now. Still open for discussion though. 🙂

**Comment by RaTcHeT302** [comment]
i think this is pretty lame, i could use something like this, i had some pretty general \ broad feedback, and it was all closed away, it kinda puts me off from honestly bringing up any suggestions anymore, and i know how valuable feedback is, from a person who has used a piece of software, for the very first time, but i find it pretty dissapointing, when some of my thoughts get tucked away, while i see the leads, or other people, being able to freely share their thoughts, and to not have their ideas hidden away somewhere

i think it's a shame, because godot has a ton of minor usability problems, which i've spotted
![](https://i.imgur.com/H5G5rYd.png)

i was even going to propose, renaming the "spatial" stuff to 3D, and a whole other buncha things, but i guess realistically there's no real interest in that, from the most experienced users, whch is kinda dissapointing

**Comment by willnationsdev** [comment]
> i was even going to propose, renaming the "spatial" stuff to 3D, and a whole other buncha things, but i guess realistically there's no real interest in that, from the most experienced users, whch is kinda dissapointing

Actually, [reduz proposed exactly that](https://github.com/godotengine/godot/issues/30736) not too long ago. And I believe that if you'd opened an Issue mentioning that suggestion, it's very likely it would have been received in a similar vein.

> i've decided to just, gather up all these ideas, and i'll keep them to myself, and if one day i become experienced enough, i'll just apply the changes to godot myself if i can

To each their own, but I always encourage people to share ideas. While some may be rejected, there is every possibility that some will be accepted. The more experience you get in understanding the project's requirements/goals, the more likely it is that suggestions will be accepted. And getting that experience takes time.

> i mostly end up feeling like nobody takes me seriously

I have never known anyone on the Godot dev team to not take a proposal seriously. What I have seen is developers accepting or rejecting proposals based on their quality level and/or suitability for the engine repository. Of course, without me reviewing your past submissions, I can't really comment on this point effectively.

> a lot of elitism pop up whenever more major projects start to grow

From what I have observed at least, most accusations of elitism I have seen appear to be misinterpretations of comments, like core devs > contributors or contributors > users. However, every time I see this happening, it is because the contributors in question simply understand the requirements of the engine repository better, and so their arguments are better grounded in the project goals. If a user likewise made suggestions that matched well with the project goals, then they would be equally valued. That is, the perceived elitism/favoritism seems to be a result of those people having a better awareness of the project overall, not because they're subjectively better somehow.

> i think godot is cool, but it's not as accessible or as open as it could be, it kinda pushes me off away personally anyway

I'm sorry to hear that. I say all this only because I believe that everyone can eventually form effective suggestions as they improve their understanding of the project goals, including you (course, I say that without having examined your submission history, so take as you will).

I myself have had *huge* amounts of work done and rejected, along with many ideas thrown out. But over the past 3-4 years of contributing, I've gradually gotten better at understanding things and forming better suggestions. And I'm constantly learning/improving in this regard. No one instantly gets that experience. Some people gain that experience faster than others, or can even pull in development experience in other fields to bolster their ability to pick up Godot's project design.

Anyway, take this as you will. I'm just a random contributor voicing his thoughts. :-) I look forward to seeing you around, whenever.

**Comment by Xrayez** [comment]
> I myself have had huge amounts of work done and rejected, along with many ideas thrown out. But over the past 3-4 years of contributing, I've gradually gotten better at understanding things and forming better suggestions. And I'm constantly learning/improving in this regard.

So far I've noticed a certain spectrum to this:

1. At the start of the proposal spectrum, the feature/limitation is obvious, so there's no point to propose something in the first place, as contributors are likely already working on this.

2. At the end of the spectrum, the feature/limitation is only relevant for specific project implementation, so it's pointless to propose this stuff either (for core).

The proposal is only relevant if it adds something in a way that most people don't see/realize yet and which would elicit the "aha" moment for many (effectively being in the middle of the proposal spectrum) but:

> i've decided to just, gather up all these ideas, and i'll keep them to myself, and if one day i become experienced enough, i'll just apply the changes to godot myself if i can

which, depending on how you're likely to contribute to open-source projects, makes the whole proposal spectrum useless. This is why Godot needs better plugin/module/gdnative support (and better advertisement of it), so that people are less likely to derail/branch off like that.
